flow/FlowUtilities.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.22;
+
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import { ud21x18, UD21x18 } from "@prb/math/src/UD21x18.sol";
+
+/// @dev A utility library to calculate the rate per second for a given amount of tokens over a specific duration, and
+/// the amounts streamed over various periods of time.
+library FlowUtilities {
+    /// @notice This function calculates the rate per second for a given amount of tokens for a specific duration.
+    /// @dev The rate per second is a 18 decimal fixed-point number and it is calculated as `amount / duration`.
+    /// @param token The address of the token.
+    /// @param amount The amount of tokens, denoted in token's decimals.
+    /// @param duration The duration in seconds wished to stream.
+    /// @return ratePerSecond The rate per second as a fixed-point number.
+    function ratePerSecondWithDuration(
+        address token,
+        uint128 amount,
+        uint40 duration
+    )
+        internal
+        view
+        returns (UD21x18 ratePerSecond)
+    {
+        // Get the decimals of the token.
+        uint8 decimals = IERC20Metadata(token).decimals();
+
+        // If the token has 18 decimals, we can simply divide the amount by the duration as it returns a 18 decimal
+        // fixed-point number.
+        if (decimals == 18) {
+            return ud21x18(amount / duration);
+        }
+
+        // Calculate the scale factor from the token's decimals.
+        uint128 scaleFactor = uint128(10 ** (18 - decimals));
+
+        // Multiply the amount by the scale factor and divide by the duration.
+        ratePerSecond = ud21x18(scaleFactor * amount / duration);
+    }
+
+    /// @notice This function calculates the rate per second for a given amount of tokens for a specific range of time.
+    /// @dev The rate per second is a 18 decimal fixed-point number and it is calculated as `amount / (end - start)`.
+    /// @param token The address of the token.
+    /// @param amount The amount of tokens, denoted in token's decimals.
+    /// @param start The start timestamp.
+    /// @param end The end timestamp.
+    /// @return ratePerSecond The rate per second as a fixed-point number.
+    function ratePerSecondForTimestamps(
+        address token,
+        uint128 amount,
+        uint40 start,
+        uint40 end
+    )
+        internal
+        view
+        returns (UD21x18 ratePerSecond)
+    {
+        // Get the decimals of the token.
+        uint8 decimals = IERC20Metadata(token).decimals();
+
+        // Calculate the duration.
+        uint40 duration = end - start;
+
+        if (decimals < 18) {
```

Comment:
Should be `==`

Reply:
https://github.com/sablier-labs/examples/pull/37/commits/9fedb7ad0f2ce57842e8b6e3da3999c39b935e7d